### ISSUE TITLE
vCard PHOTOs are small

### ISSUE THREAD

**Issue by kocjs** [SimpleMobileTools/Simple-Contacts#533]
Thank you for this great app!

MotoX4 lineage os 16.1 

Expected behavior:

Export to .vcf preserves the full quality of the image.

Actual behavior:

Exportred contacts have a low resolution image only.

**Comment by kocjs** [comment]
An other app MC backup can do this, it uses the Type=PNG wile exporting.
`PHOTO;TYPE=PNG;ENCODING=B:/`

**Comment by tibbi** [comment]
exporting photos as pngs sounds like a terrible idea

**Comment by kocjs** [comment]
I am just a user, so no idea. but they are not alone. 

> BEGIN:VCARD
VERSION:3.0
PRODID:-//Thunderbird.net/NONSGML Thunderbird CardBook V43.8//EN-US
UID:147cf6d6-aabe-4ce4-9dc7-4f52e5bbc011
FN:test
N:test;;;;
TEL:93457
REV:2020-04-14T20:34:10Z
PHOTO;ENCODING=B;TYPE=PNG:iVBORw0KGgoAAAANSUhEUgAAB4AAAAQ4CAYAAADo08FDAAAAB

> BEGIN:VCARD
VERSION:3.0
UID:pas-id-3f2aa1b3a033a82611216f4dd6712f2d2fd97074
X-EVOLUTION-FILE-AS:test
FN:test
N:;test;;;
TEL;X-EVOLUTION-E164=3270507711,"+91";TYPE=CELL:009-132-70507711
FN:test
N:;test;;;
TEL;X-EVOLUTION-E164=3270507711,"+91";TYPE=CELL:00913270507711
PHOTO;TYPE=PNG;ENCODING=b:/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAEBAQEBAQEBAQEBA

**Comment by tibbi** [comment]
I guess these are not real photos, just some small avatars or so

**Comment by kocjs** [comment]
No, those vCard are 2.3 mb each and the carry a picture, to be exact it was just a screen shot to test, that's where this png comes form. On the phone i use an app that gets the whatsup profile picture. That's where those png probably come form. However if i use a jpeg it results like this.

> BEGIN:VCARD
VERSION:3.0
PRODID:-//Thunderbird.net/NONSGML Thunderbird CardBook V43.8//EN-US
UID:5bff8a7f-69ee-4c5f-a112-2a5b07e17718
FN:test-tp
N:;;;;
TEL:666
REV:2020-04-17T20:30:30Z
PHOTO;ENCODING=B;TYPE=JPEG:/9j/4QaHRXhpZgAATU0AKgAAAAgADwE7AAIAAAAFAAAAwgEP

**Comment by kocjs** [comment]
Are there any issues with png in general? Or is it just the size?

**Comment by tibbi** [comment]
the size, real photos should never be saved as pngs. Are those contacts saved into the "Phone storage (not visible by other apps)" contact source, or some other casual synced device contact source, like your email or so?

**Comment by kocjs** [comment]
The contacts are saved in an address book provided by DecSynccc

**Comment by rezad1393** [comment]
> the size, real photos should never be saved as pngs. 

why? whe are no longer is the smartphone age. we are in iOS and android age.
I dont think reducing image to a blurry 4kB picture is very bad and a bug.
other apps (That are not open source of couse) save the image (or a smaller but not that bad version of image)

**Comment by rezad1393** [comment]
another option would be to ask user what he wants at the export step.
I dont know how android saves contact images but android OS clearly doesn't feel justified to reduce image quality a lot.
so the OS doesn't think it needs to do that. then user sees a clear image for its contact. if a contact backup destroys the image quality this bad, then changing this is not a feature request, it is a bug fix request.

I have no idea why the creator of this app choose to do it this way, he had his own reasons.
but keeping it this way is a dark mark for this app as it would messes with the most important thing for user (user data) so bad that  would cause other users to discourage potential users from this app.

**Comment by Charles7z** [comment]
This is a pain for anyone who wants to use this app without cloud storage.

You might as well delete all contact photos cause they will look like sh!t every time you hand to restore. At least the big ugly letter next to each contact from then on is clear 🤣

**Comment by Charles7z** [comment]
I tested several apps to backup contacts and keep reasonable image size. The only one that worked is super backup:

https://play.google.com/store/apps/details?id=com.idea.backup.smscontacts

**Comment by ravenkurisu** [comment]
just wanted to bump this thread, I would like to have the backups save original sizes. I don't mind the extra space it requires.
